### ISSUE TITLE
bcf::Record::info should not take `&mut self`

### ISSUE THREAD

**Issue by holtgrewe** [rust-bio/rust-htslib#78]
Rather, there should be a mut and non-mut `Info` type similar to how there is `iter()` and `iter_mut()`.

@johanneskoester what do you think?

**Comment by johanneskoester** [comment]
Yes, in particular since we discovered that htslib is copying the info information out of the record object, right? So the whole mutable Info functionality is obsolete? Or am I missing something?

**Comment by felix-clark** [comment]
By my reading of the docs, the `Info` struct does not expose any mutable references and its interface is functionally read-only. The INFO field of the BCF data is modified instead through the `clear_info_*()` and `push_info_*()` functions of `Record`. Unless there is some other issue in the implementation it looks like `Info`'s public functions, along with its private `data()` function, can be changed to take immutable references to `self`.

**Comment by felix-clark** [comment]
This appears to have been dealt with since version 0.35. Just wanted to point out that this issue can probably be closed.
